### DISCUSSION TITLE
Google Analytics Tracking

### DISCUSSION THREAD

**Discussion by bizharvest** [add2cal/add-to-calendar-button#489] [Q&A] [answered]
I'm a bit confused on how event clicks are being tracked. Is this automatically configured? Do I need to setup a listener? Do you have a good step by step instruction/example on how this is implemented?

The other questions that was similar was not very helpful and the docs talk about tracking but there is no good clear example available that I could find.

**Comment by jekuer** [comment, marked as answer]
Hi @bizharvest ,

The easiest way with GA is using the dataLayer. We push the following events there:

- initialization
- openList
- closeList
- openCalendarLink
- openSingletonLink
- openSubEventLink
- success

However, you might need to use Google Tag Manager for this. I am not sure whether it works natively with pure GA4 (I guess it should).
See Google's documentation for more details: https://developers.google.com/tag-platform/tag-manager/datalayer?hl=en
Or search results like this: https://www.tagmate.app/blogs/track-datalayer-enabled-form-in-google-analytics-4-step-by-step-guide

You can also observe the button via its atcb-last-event attribute as described at https://add-to-calendar-button.com/advanced-use#case-7.
